### PR TITLE
fix: dataset layers are not exported correctly

### PR DESCRIPTION
**Pull request by mimoham24**
# Overview
fixing the layer exporting bug
## What I've done
encoder now fetches the property values first, if no values found -> fetch values from dataset

## How I tested
export all types "KML, CZML, GeoJSON, SHP"

### CODE REVIEW

File: pkg/layer/encoding/czml.go [hunk 1]
```diff
@@ -56,13 +56,26 @@ func (e *CZMLEncoder) encodeSingleLayer(li *merging.SealedLayerItem) (*czml.Feat
 		if li.Property.Field("location") != nil {
```
Comment (rot1024) [inline pkg/layer/encoding/czml.go:56]:
```suggestion
		if f := li.Property.Field("location"); f != nil {
```

More readable code

Change other parts of the file in the same way.

File: pkg/layer/encoding/czml.go [hunk 2]
```diff
@@ -56,13 +56,26 @@ func (e *CZMLEncoder) encodeSingleLayer(li *merging.SealedLayerItem) (*czml.Feat
...
 			latlng, ok = li.Property.Field("location").PropertyValue.ValueLatLng()
```
Comment (rot1024) [inline pkg/layer/encoding/czml.go:57]:
```suggestion
			latlng, ok = f.PropertyValue.ValueLatLng()
```

File: pkg/layer/encoding/czml.go [hunk 3]
```diff
@@ -56,13 +56,26 @@ func (e *CZMLEncoder) encodeSingleLayer(li *merging.SealedLayerItem) (*czml.Feat
...
 			if !ok {
-				return nil, errors.New("invalid value type")
+				dsll := li.Property.Field("location").DatasetValue.ValueLatLng()
```
Comment (rot1024) [inline pkg/layer/encoding/czml.go:59]:
```suggestion
				dsll := f.DatasetValue.ValueLatLng()
```

**Review by rot1024 — DISMISSED** [review]
Changes are ok but I've tried to make the code a little more readable.

**Review by HideBa — APPROVED** [review]
Thanks! 👍

**Review by HideBa — APPROVED** [review]

**Review by rot1024 — APPROVED** [review]

**Pull request merged**